packages/stores/src/account/base.ts

```diff
@@ -749,4 +764,101 @@ export class AccountStore<Injects extends Record<string, any>[] = []> {
       sequence: Number(account.sequence.toString()),
     };
   }
+
+  public async estimateFee(
+    wallet: ChainWalletBase,
+    messages: readonly EncodeObject[],
+    fee: Optional<TxFee, "gas">,
+    memo: string
+  ): Promise<TxFee> {
+    const encodedMessages = messages.map((m) => this.registry.encodeAsAny(m));
+    const { sequence } = await this.getSequence(wallet);
+
+    const unsignedTx = TxRaw.encode({
+      bodyBytes: TxBody.encode(
+        TxBody.fromPartial({
+          messages: encodedMessages,
+          memo: memo,
+        })
+      ).finish(),
+      authInfoBytes: AuthInfo.encode({
+        signerInfos: [
+          SignerInfo.fromPartial({
+            // Pub key is ignored.
+            // It is fine to ignore the pub key when simulating tx.
+            // However, the estimated gas would be slightly smaller because tx size doesn't include pub key.
+            modeInfo: {
+              single: {
+                mode: SignMode.SIGN_MODE_LEGACY_AMINO_JSON,
+              },
+              multi: undefined,
+            },
+            sequence,
+          }),
+        ],
+        fee: Fee.fromPartial({
+          amount: fee.amount.map((amount) => {
+            return { amount: amount.amount, denom: amount.denom };
+          }),
+        }),
+      }).finish(),
+      // Because of the validation of tx itself, the signature must exist.
+      // However, since they do not actually verify the signature, it is okay to use any value.
+      signatures: [new Uint8Array(64)],
+    }).finish();
+
+    const restEndpoint = getEndpointString(await wallet.getRestEndpoint(true));
```

Comment:
One thing I noticed with Keplr's work that we may want to handle (depending on if we want to simulate IBC transactions on counterparty chains, likely yes), is they check the version of the Cosmos SDK being used for simulating the fee.

We could either do that check, or just accept a failed request to handle that case. (the latter is fine with me)

Reply:
Agreed. I'm actually using [Keplr's check](https://github.com/chainapsis/keplr-wallet/blob/93a13dcceb91bea88dd63934d2fa8d1a234ab38c/packages/hooks/src/tx/gas-simulator.ts#L410-L418) to determine if we need the [backup fee](https://github.com/osmosis-labs/osmosis-frontend/blob/d6dd1c45a9de6b13d632732a2d8bb88f68b1624c/packages/stores/src/account/base.ts#L876-L878)